uwblueprint/headsupguys
# Tool Home Page

## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Tool-Builder-Form-5a6b5a9e8edb4c13ba8f1e43adeebd0e)

[Deploy Preview](https://deploy-preview-33--modest-payne-4c3f5b.netlify.app/create-a-tool)


## Implementation description
* Admin use the Tool Builder to dynamically generate landing pages for each Tool as well as the Tool Self Check. It primarily consists of a form in which values are provided for elements of Tool landing pages which vary based on each Tool.



## Steps to test
1. Compare to Design to Figma > Builders > Create a Tool
https://www.figma.com/file/GhDd9C0XsyVDgwL5zzvDvO/HUG-S21?node-id=1059%3A16927
2. Ensure that changes to input fields persist when switching between the home and self check pages
3. Ensure that the "Save" Button does not allow you to save an empty document or the last saved document
4. Endure that submit is only allowed when all required fields are filled and that pointing at the greyed out submit button tells you the correct field that needs to be filled out next
5. Ensure that the objects that are printed out when saved/publish match the information you put in
6. Ensure that on the self check question page, you are unable ONLY numbers in the corresponding values AFTER clicking the "123" button
7. Type some information into a multiple choice / multi-select question, change the question type to a short/long answer question and ensure that your original information persists upon switching back. (This will not be the case for the slider as it needs to override the "value" field to keep track of whether the slider should start at 0 or 1)
8. Ensure that adding and moving around questions all over the place produces no errors in the web console.
9. Ensure that the discard clears both the home page and  self check as expected
10. Ensure that the option removal, delete buttons + question button and (Tool Home Page) add link removal buttons work as intended.


## What should reviewers focus on?
* Design
* Functionality


## Checklist
